components/application-registry/internal/metadata/applications/util.go

```diff
@@ -138,30 +138,55 @@ func replaceService(id string, app *v1alpha1.Application, service v1alpha1.Servi
 	}
 }
 
-func ensureServiceExists(id string, app *v1alpha1.Application) apperrors.AppError {
-	if !serviceExists(id, app) {
+func checkServiceCanBeReplaced(id, displayName string, app *v1alpha1.Application) apperrors.AppError {
+	if !serviceExistsWithId(id, app) {
 		message := fmt.Sprintf("Service with ID %s does not exist", id)
 
 		return apperrors.NotFound(message)
 	}
 
+	if !serviceExistsWithNormalizedDisplayName(displayName, app) {
+		message := fmt.Sprintf("Cannot change service name to %s for service with ID %s", displayName, id)
+
+		return apperrors.WrongInput(message)
+	}
+
 	return nil
 }
 
-func ensureServiceNotExists(id string, app *v1alpha1.Application) apperrors.AppError {
-	if serviceExists(id, app) {
+func checkServiceCanBeAdded(id, displayName string, app *v1alpha1.Application) apperrors.AppError {
+	if serviceExistsWithId(id, app) {
 		message := fmt.Sprintf("Service with ID %s already exists", id)
 
 		return apperrors.AlreadyExists(message)
 	}
 
+	if serviceExistsWithNormalizedDisplayName(displayName, app) {
+		message := fmt.Sprintf("Service with name %s already exists", displayName)
+
+		return apperrors.WrongInput(message)
+	}
+
```

Comment:
I don't understand why we return `apperrors.WrongInput()` and the reason is that the service already exists. The message suggests the same.

Reply:
That's how new logic works. User should not be allowed to register the service with displayName that is not unique. This is considered as wrong input data (Status bad request 400)